### DISCUSSION TITLE
Window movement , placement and resizing

### DISCUSSION THREAD

**Discussion by saeedtahmtan** [labwc/labwc#1573] [Q&A]
Hi everyone,

I'm working on a fork of this project and I'm interested in adding animations. I'm hoping to use the **Lerp** function to achieve smooth window movement and resizing. Could anyone provide guidance on how to implement this?

The animation should continue until the window reaches the desired size and position.

**Comment by johanmalm** [comment]
I ~~can~~ cannot speak with any authority on this because I haven't got any experience with it. 

However, I image one approach would be to:

1. Add a timer. I guess 50 ms or something of that sort of duration. See example here: https://github.com/labwc/labwc/blob/9b47a5e11ebc88624fbfdb9c889cbae78cfbb0e5/src/workspaces.c#L251
2. Add an `advance()` method to `view_impl`. https://github.com/labwc/labwc/blob/9b47a5e11ebc88624fbfdb9c889cbae78cfbb0e5/include/view.h#L90
3. Call `view->impl->advance()` (probably using a `view_advance()` wrapper) every time the timer expires
4. In the `advance()` function, apply the Lerp function using `view->pending.{x,y}`

**Reply by saeedtahmtan** [reply]
I don't think using a timer is a good option. I added a ```wlr_box``` view struct as the "target". As far as I checked the source, it has a ```view_move``` function, and every time this function is executed, I need to move x and y closer to the target!
But it not continue to reach the target :( .

**Reply by saeedtahmtan** [reply]
Since I am a game developer, let me give an example to illustrate my point. In games, we have a loop that updates the state for us each time. Each time this function is executed, it calculates the time elapsed since the previous time and applies it to the changes to ensure smooth movement. However, here I am dealing with a different form of information and updates, and it seems that input events are being used.

[For Example](https://codepen.io/nanonansen/pen/oRWmaY)

**Comment by johanmalm** [comment]
It might be worth looking at the wayfire code, or reaching out to them directly to seek advice on approach options.
